contracts/margin/external/BucketLender/BucketLender.sol

```diff
@@ -0,0 +1,1242 @@
+            loanOffering.rates.maxAmount == MathHelpers.maxUint256(),
+            "BucketLender#verifyLoanOffering: loanOffering.maxAmount is incorrect"
+        );
+        require(
+            loanOffering.rates.minAmount == 0,
+            "BucketLender#verifyLoanOffering: loanOffering.minAmount is non-zero"
+        );
+        require(
+            loanOffering.rates.minHeldToken == 0,
+            "BucketLender#verifyLoanOffering: loanOffering.minHeldToken is non-zero"
+        );
+        require(
+            loanOffering.rates.lenderFee == 0,
+            "BucketLender#verifyLoanOffering: loanOffering.lenderFee is non-zero"
+        );
+        require(
+            loanOffering.rates.takerFee == 0,
+            "BucketLender#verifyLoanOffering: loanOffering.takerFee is non-zero"
+        );
+        require(
+            loanOffering.expirationTimestamp == MathHelpers.maxUint256(),
+            "BucketLender#verifyLoanOffering: expirationTimestamp is incorrect"
+        );
+        require(
+            loanOffering.salt == 0,
+            "BucketLender#verifyLoanOffering: loanOffering.salt is non-zero"
+        );
+
+        // CHECK VALUES32
+        require(
+            loanOffering.callTimeLimit == CALL_TIMELIMIT,
+            "BucketLender#verifyLoanOffering: loanOffering.callTimelimit is incorrect"
+        );
+        require(
+            loanOffering.maxDuration == MAX_DURATION,
+            "BucketLender#verifyLoanOffering: loanOffering.maxDuration is incorrect"
+        );
+        assert(loanOffering.rates.interestRate == INTEREST_RATE);
+        assert(loanOffering.rates.interestPeriod == INTEREST_PERIOD);
+
+        // no need to require anything about loanOffering.signature
+
+        return address(this);
+    }
+
+    /**
+     * Called by the Margin contract when anyone transfers ownership of a loan to this contract.
+     * This function initializes this contract and returns this address to indicate to Margin
+     * that it is willing to take ownership of the loan.
+     *
+     * @param  from        Address of the previous owner
+     * @param  positionId  Unique ID of the position
+     * @return             This address on success, throw otherwise
+     */
+    function receiveLoanOwnership(
+        address from,
+        bytes32 positionId
+    )
+        external
+        onlyMargin
+        nonReentrant
+        onlyPosition(positionId)
+        returns (address)
+    {
+        MarginCommon.Position memory position = MarginHelper.getPosition(DYDX_MARGIN, POSITION_ID);
+        uint256 initialPrincipal = position.principal;
+        uint256 minHeldToken = MathHelpers.getPartialAmount(
+            uint256(MIN_HELD_TOKEN_NUMERATOR),
+            uint256(MIN_HELD_TOKEN_DENOMINATOR),
+            initialPrincipal
+        );
+
+        assert(initialPrincipal > 0);
+        assert(principalTotal == 0);
+        assert(from != address(this)); // position must be opened without lending from this position
```

Comment:
just want to confirm this should be `assert` not `require`

Reply:
yes due to the first line in `verifyLoanOffering()`